### PR TITLE
[DLS-8492] Fixing Cya Navigation for small producer edit and some nav…

### PR DESCRIPTION
**Pull request by JakeReid2020**
…igation for remove

### CODE REVIEW

File: conf/app.routes [hunk 1]
```diff
@@ -99,11 +99,15 @@ POST       /return-change-registration                          controllers.Retu
 
 GET        /remove-small-producer-confirm/:sdil                 controllers.RemoveSmallProducerConfirmController.onPageLoad(mode: Mode = NormalMode, sdil: String)
 POST       /remove-small-producer-confirm/:sdil                 controllers.RemoveSmallProducerConfirmController.onSubmit(mode: Mode = NormalMode, sdil: String)
+GET        /remove-small-producer-confirm/:sdil                 controllers.RemoveSmallProducerConfirmController.onPageLoad(mode: Mode = CheckMode, sdil: String)
```
Comment (drewmiley) [inline conf/app.routes:102]: Are these duplicates of the one above - do we need a slightly different route name?

File: conf/app.routes [hunk 2]
```diff
@@ -99,11 +99,15 @@ POST       /return-change-registration                          controllers.Retu
...
+POST       /remove-small-producer-confirm/:sdil                 controllers.RemoveSmallProducerConfirmController.onSubmit(mode: Mode = CheckMode, sdil: String)
 
 GET        /add-small-producer                                  controllers.AddASmallProducerController.onPageLoad(mode: Mode = NormalMode)
 POST       /add-small-producer                                  controllers.AddASmallProducerController.onSubmit(mode: Mode = NormalMode)
 GET        /add-small-producer-edit                             controllers.AddASmallProducerController.onEditPageLoad(mode: Mode = EditMode, sdilReference: String)
 POST       /add-small-producer-edit                             controllers.AddASmallProducerController.onEditPageSubmit(mode: Mode = EditMode, sdilReference: String)
+GET        /add-small-producer-edit                             controllers.AddASmallProducerController.onEditPageLoad(mode: Mode = CheckMode, sdilReference: String)
```
Comment (drewmiley) [inline conf/app.routes]: Are these duplicates of the one above - do we need a slightly different route name?

File: app/views/helpers/returnDetails/PackAtBusinessAddressSummary.scala [hunk 3]
```diff
@@ -30,7 +30,7 @@ object PackAtBusinessAddressSummary {
   def summaryList(userAnswers: UserAnswers, isCheckAnswers: Boolean)
                           (implicit messages: Messages): Option[SummaryList] = {
     userAnswers.get(PackAtBusinessAddressPage) match {
-      case Some(true) =>
+      case Some(a) =>
```
Comment (drewmiley) [inline app/views/helpers/returnDetails/PackAtBusinessAddressSummary.scala]: Is this change intended? Looks a little unusual

File: app/views/helpers/returnDetails/SmallProducerDetailsSummary.scala [hunk 4]
```diff
@@ -52,9 +52,9 @@ object SmallProducerDetailsSummary extends SummaryListRowLitresHelper with Retur
         key     = smallProducer.sdilRef,
         value   = value,
         actions = Seq(
-          ActionItemViewModel("site.edit", routes.AddASmallProducerController.onEditPageLoad(smallProducer.sdilRef).url)
+          ActionItemViewModel("site.edit", routes.AddASmallProducerController.onEditPageLoad( if (mode == NormalMode) {EditMode} else CheckMode, smallProducer.sdilRef).url)
```
Comment (drewmiley) [inline app/views/helpers/returnDetails/SmallProducerDetailsSummary.scala:55]: Might be able to change `{EditMode}` to `EditMode` for consistency

**Review by JayenKerai — APPROVED** [review]
